### ISSUE TITLE
Support User Assigned Identities for EventHub Namespace resources

### ISSUE THREAD

**Issue by favoretti** [Azure/azure-sdk-for-go#14674]
### Feature Request

I'm working on implementing BYOK encryption support for EventHub Namespaces in AzureRM terraform provider in this PR: https://github.com/terraform-providers/terraform-provider-azurerm/pull/10821

While trying to implement it, lack of User Assigned identities makes it quite a challenge:
CURRENT ISSUE with finalizing:
- `2018-01-01-preview` EventHub API version does not support user-assigned identities.
- While using managed identity - BYOK can not be inlined into `azurerm_eventhub_namespace` resource, since namespace needs to have the rights to access the KeyVault containing the custom key.
- Implementing it as it's implemented in this PR is a no-go, since virtual resource effectively becomes "indestructible", or, in other words, one can not force namespace re-creation if BYOK ever gets removed.

POTENTIAL SOLUTION:
- Introduce support for user-assigned identity in EventHub namespaces, so that rights can be assigned pre namespace creation and BYOK can be inlined in the resource.

Would be awesome if Service Team could consider implementing User Assigned Identities for EH Namespaces.

**Comment by ArcturusZhang** [comment]
Hi @favoretti thanks for this issue!

I have once worked on the same work item, and it turned out to be blocked by a wrong schema in swaggers.
Eventhub has support for identities in this api-version: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/eventhub/resource-manager/Microsoft.EventHub/preview/2021-01-01-preview/namespaces-preview.json#L601
This is not released in azure-sdk-for-go yet, but now it does not make sense to release in the SDK because the definition of `identity` is wrong. 
ARM has a standard that the `identity` must in the root level of a resource's creation request body, but in this case, you will find out that the identity is defined under `properties` of the namespace resource payload. Therefore it is wrong and will not work. I have tried that quite a time ago and reported this to the corresponding service team. I did not get a response until now.

FYI exactly the same issue exists for `servicebus` too.
